specification/confidentialledger/resource-manager/Microsoft.ConfidentialLedger/preview/2023-01-26-preview/confidentialledger.json

```diff
@@ -0,0 +1,708 @@
+          "description": "Public key of the user cert (.pem or .cer)",
+          "type": "string"
+        },
+        "ledgerRoleName": {
+          "$ref": "#/definitions/LedgerRoleName"
+        }
+      }
+    },
+    "AADBasedSecurityPrincipal": {
+      "type": "object",
+      "description": "AAD based security principal with associated Ledger RoleName",
+      "properties": {
+        "principalId": {
+          "description": "UUID/GUID based Principal Id of the Security Principal",
+          "type": "string"
+        },
+        "tenantId": {
+          "description": "UUID/GUID based Tenant Id of the Security Principal",
+          "type": "string"
+        },
+        "ledgerRoleName": {
+          "$ref": "#/definitions/LedgerRoleName"
+        }
+      }
+    },
+    "LedgerProperties": {
+      "type": "object",
+      "description": "Additional Confidential Ledger properties.",
+      "title": "LedgerProperties",
+      "properties": {
+        "ledgerName": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Unique name for the Confidential Ledger."
+        },
+        "ledgerUri": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Endpoint for calling Ledger Service."
+        },
+        "identityServiceUri": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Endpoint for accessing network identity."
+        },
+        "ledgerInternalNamespace": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Internal namespace for the Ledger"
+        },
+        "ledgerType": {
+          "description": "Type of Confidential Ledger",
+          "$ref": "#/definitions/ConfidentialLedgerType"
+        },
+        "provisioningState": {
+          "description": "Provisioning state of Ledger Resource",
+          "readOnly": true,
+          "$ref": "#/definitions/ProvisioningState"
+        },
+        "aadBasedSecurityPrincipals": {
+          "description": "Array of all AAD based Security Principals.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AADBasedSecurityPrincipal"
+          },
+          "x-ms-identifiers": [
+            "principalId"
+          ]
+        },
+        "certBasedSecurityPrincipals": {
+          "description": "Array of all cert based Security Principals.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CertBasedSecurityPrincipal"
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "ConfidentialLedger": {
+      "type": "object",
+      "description": "Confidential Ledger. Contains the properties of Confidential Ledger Resource.",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/TrackedResource"
+        },
+        {
+          "$ref": "#/definitions/ResourceLocation"
+        },
```

Comment:
To fix this you could remove the ResourceLocation or set its location as required.

The property `location` is defined in both TrackedResource and ResourceLocation. But the definition has conflict with one other cause one is required but the other is optional.

https://github.com/Azure/azure-rest-api-specs/blob/d883967b5cc43f9a7fb9c53d0fb3c36bcc373da9/specification/common-types/resource-management/v3/types.json#L73-L84


https://github.com/Azure/azure-rest-api-specs/blob/d883967b5cc43f9a7fb9c53d0fb3c36bcc373da9/specification/confidentialledger/resource-manager/Microsoft.ConfidentialLedger/preview/2023-01-26-preview/confidentialledger.json#L421-L429

Reply:
Thanks for the pointer. I removed `ResourceLocation` and JS is now building.